contrib/platform/src/com/sun/jna/platform/win32/Advapi32.java

```diff
@@ -1635,6 +1635,144 @@ boolean GetFileSecurity(String lpFileName,
                             IntByReference lpnLengthNeeded);
 
     /**
+     * The SetFileSecurity function sets the security of a file or directory object.
+     * This function is obsolete. Use the SetNamedSecurityInfo function instead.
+     *
+     * @param lpFileName
+     *          A pointer to a null-terminated string that specifies the file or directory for which security is set.
+     *          Note that security applied to a directory is not inherited by its children.
+     * @param SecurityInformation
+     *          Specifies a SECURITY_INFORMATION structure that identifies the contents of the security descriptor
+     *          pointed to by the pSecurityDescriptor parameter.
+     * @param pSecurityDescriptor
+     *          A pointer to a SECURITY_DESCRIPTOR structure.
+     * @return
+     *          If the function succeeds, the function returns nonzero. If the function fails, it returns zero. To get
+     *          extended error information, call GetLastError.
+     */
+    boolean SetFileSecurity(String lpFileName, int SecurityInformation, Pointer pSecurityDescriptor);
```

Comment:
Please check if pSecurityDescriptor could be declared to take a WinNT.SECURITY_DESCRIPTOR. I'm not sure how sensible this is, as SECURITY_DESCRIPTOR is basicly an opaque strucure, but it would be typed that way (maybe that structure needs a constructor added that takes an allocation size).